### PR TITLE
Add strings for the intro-pricing-end countdown

### PR DESCRIPTION
**Pull request by Vinnl**

### CODE REVIEW

File: en/app.ftl [hunk 1]
```diff
@@ -159,6 +159,18 @@ whatsnew-feature-tracker-removal-snippet = Now { -brand-name-relay } can remove
...
+# When translating, please make sure the resulting string is of roughly similar
+# length as the English version.
+# Variables:
+#   $end_date (string) - The localised date the introductory pricing offer ends, e.g. "September 27 2022"
+whatsnew-feature-offer-countdown-snippet = Get { -brand-name-relay-premium } before { $end_date } and enjoy unlimited masking at our…
+# Variables:
+#   $end_date (string) - The localised date the introductory pricing offer ends, e.g. "September 27 2022"
+whatsnew-feature-offer-countdown-description = Get { -brand-name-relay-premium } before { $end_date } and enjoy unlimited masking at our intro month-to-month price.
+whatsnew-feature-offer-countdown-cta = Upgrade now
+
```
Comment (Vinnl) [inline en/app.ftl:173]: image

image

File: en/app.ftl [hunk 2]
```diff
@@ -198,6 +210,12 @@ landing-hero-headline-2 = Protect your true email address to help control your i
 landing-hero-body-2 = { -brand-name-firefox-relay } email masks protect your true email address from public view, automatically forwarding messages to your true inbox.
     Now you can receive only the messages you want in your inbox. Sign up with your { -brand-name-firefox-account } to get started.
 
+landing-offer-end-hero-heading = Our intro pricing offer is ending soon!
+landing-offer-end-hero-cta = Upgrade now
+# Variables:
+#   $end_date (string) - The localised date the introductory pricing offer ends, e.g. "September 27 2022"
+landing-offer-end-hero-content = Get { -brand-name-relay-premium } before { $end_date } and enjoy premium email protection at our intro month-to-month price.
+
```
Comment (Vinnl) [inline en/app.ftl:218]: image

File: en/app.ftl [hunk 3]
```diff
@@ -245,6 +263,15 @@ landing-pricing-premium-feature-6 = Email tracker removal
...
 
+# Variables:
+#   $monthly_price (string) - the monthly cost (including currency symbol) for Relay Premium. Examples: $0.99, 0,99 €
+landing-pricing-offer-end-headline = Introductory price: Unlimited masks for { $monthly_price } per month
+landing-pricing-offer-end-warning = This promo is expiring soon
+landing-pricing-offer-end-cta = Upgrade now
+# Variables:
+#   $end_date (string) - The localised date the introductory pricing offer ends, e.g. "September 27 2022"
+landing-pricing-offer-end-body = Get { -brand-name-relay-premium } before { $end_date } and enjoy premium email protection at our intro month-to-month price.
+
```
Comment (Vinnl) [inline en/app.ftl:274]: image

File: en/app.ftl [hunk 4]
```diff
@@ -316,7 +343,17 @@ premium-promo-hero-headline = Make protecting your inbox easier with { -brand-na
...
+#   $monthly_price (string) - the monthly cost (including currency symbol) for Relay Premium. Examples: $0.99, 0,99 €
+premium-promo-hero-body-3 = With { -brand-name-firefox-relay-premium }, you get unlimited custom email masks that forward only the emails you want to your true email address.
 premium-promo-hero-cta = Upgrade now
+
+premium-promo-offer-end-hero-heading = Our intro pricing offer is ending soon!
+# Variables:
+#   $end_date (string) - The localised date the introductory pricing offer ends, e.g. "September 27 2022"
+premium-promo-offer-end-hero-content = Get { -brand-name-relay-premium } before { $end_date } and enjoy premium email protection at our intro month-to-month price.
+premium-promo-offer-end-hero-cta = Upgrade now
+
```
Comment (Vinnl) [inline en/app.ftl:356]: image

File: en/app.ftl [hunk 5]
```diff
@@ -347,6 +384,15 @@ premium-promo-use-cases-gaming-body-2 = Worried about other gamers or trolls lea
...
 
+# Variables:
+#   $monthly_price (string) - the monthly cost (including currency symbol) for Relay Premium. Examples: $0.99, 0,99 €
+premium-promo-pricing-offer-end-headline = Introductory price: Unlimited masks for { $monthly_price } per month
+premium-promo-pricing-offer-end-warning = This promo is expiring soon
+premium-promo-pricing-offer-end-cta = Upgrade now
+# Variables:
+#   $end_date (string) - The localised date the introductory pricing offer ends, e.g. "September 27 2022"
+premium-promo-pricing-offer-end-body = Get { -brand-name-relay-premium } before { $end_date } and enjoy premium email protection at our intro month-to-month price.
+
```
Comment (Vinnl) [inline en/app.ftl:395]: image

File: en/app.ftl [hunk 6]
```diff
@@ -698,6 +744,13 @@ banner-upgrade-loyalist-headline-2 = Protect your privacy, save the internet
 banner-upgrade-loyalist-copy-2 = Protect your privacy while joining our mission to build a better internet, all for { $monthly_price }
 banner-upgrade-loyalist-cta = Get more protection
 
+# End of intro pricing countdown banner:
+banner-offer-end-headline = Our intro pricing offer ends soon!
+# Variables:
+#   $end_date (string) - The localised date the introductory pricing offer ends, e.g. "September 27 2022"
+banner-offer-end-copy = Get { -brand-name-relay-premium } before { $end_date } and enjoy unlimited masking at our intro month-to-month price.
+banner-offer-end-cta = Upgrade now
+
```
Comment (Vinnl) [inline en/app.ftl:753]: image

File: en/app.ftl [hunk 7]
```diff
@@ -972,6 +1062,9 @@ forwarded-email-header-cc-notice-2 = { -brand-name-relay-premium } removes CCs a
 #   $premium_link (string) - This is a link to relay.firefox.com/premium. Example: <a href="https://relay.firefox.com/premium" ...>Firefox Relay Premium</a>
 forwarded-email-header-premium-banner-2 = Upgrade to { $premium_link } to get unlimited email masks and a custom email subdomain.
 # Variables:
+#   $premium_link (string) - This is a link to relay.firefox.com/premium. Example: <a href="https://relay.firefox.com/premium" ...>Relay Premium</a>
+forwarded-email-header-offer-countdown-banner = Our intro monthly pricing offer is expiring. Upgrade to { $premium_link }.
```
Comment (Vinnl) [inline en/app.ftl:1066]: image

File: en/app.ftl [hunk 8]
```diff
@@ -982,6 +1075,9 @@ forwarded-email-trackers-blocked-report = Learn more
 forwarded-email-footer-2 = Stop email forwarding and manage settings for all masks here.
 # This entire text is a link
 forwarded-email-footer-premium-banner = Upgrade to { -brand-name-relay-premium }
+# Variables:
+#   $premium_link (string) - This is a link to relay.firefox.com/premium. Example: <a href="https://relay.firefox.com/premium" ...>Relay Premium</a>
+forwarded-email-footer-offer-countdown-banner = Upgrade to { $premium_link } before our intro pricing expires.
```
Comment (Vinnl) [inline en/app.ftl:1080]: image

**Review by Vinnl — COMMENTED** [review]
Screenshots inline.

**Comment by peiying2** [comment]
@Vinnl a comment on the "comments" for the strings for Days, Minutes and Seconds. 

Questions: given the conversation we have in the Slack, are these the strings you are referring to? Promotion is only for a select few markets but not globally? If so:
- we can put these strings in a separate file, and it can be controlled in a TOML file which languages are required to localize the promo so no time is wasted for the contributors.
- Another benefit to separate the the strings from the rest is that promotion can be temporary. Once done, we can remove the string or file. This way when new language is added (per community request), they don't have to translate the file that never gets used in their language/market.

**Comment by Vinnl** [comment]
Hi @peiying2, no, these strings apply to everybody. Indeed some of these strings will no longer be visible when the countdown runs out, but I assume we can just remove them from `en/app.ftl` again then (just like we can remove the code that renders them), to avoid future translations?

(I'll keep your comment in mind for when we do get to the work that won't be available everywhere right away though.)

File: en/app.ftl [hunk 9]
```diff
@@ -924,6 +977,43 @@ modal-domain-register-success-2 = { $subdomain } is now your email subdomain!
...
+        *[other] { $remaining_hours ->
+            [0] { $remaining_days } days remaining
+            [1] { $remaining_days } days and 1 hour remaining
+            *[other] { $remaining_days } days and { $remaining_hours } hours remaining
+        }
+    }
+# This is a label displayed on top of a large number representing the number of days that the introductory pricing offer is still valid
+# There's not much room for this (about five characters), so this might need abbreviating.
+offer-countdown-timer-days = Days
+# This is a label displayed on top of a large number representing the number of hours of the remaining time the introductory pricing offer is still valid
```
[the remark this answers is not in the document]
Comment (Vinnl) [reply en/app.ftl]: Oh really? I thought this was valid English (or should I say: I thought _that_ this was valid English? :stuck_out_tongue: ) Anyway, fixed in c4898af.

[the remark this answers is not in the document]
Comment (peiying2) [reply en/app.ftl]: I ran into the same strings in the add-on repo and this sentence seemed so long. It's only a comment but our contributors to notice this type of things. Thanks for the fix.

**Review by peiying2 — APPROVED** [review]

**Comment by peiying2** [comment]
Thanks for the updates and comment. The TOML configuration allows for customized content for specific markets. Keep that in mind when the opportunity presents.  And yes, let's make sure the promotion is removed after the deadline has passed.

**Comment by peiying2** [comment]
@Vinnl is there are deadline for the promotional strings? I want to put it in Pontoon if these strings are time sensitive.

**Comment by Vinnl** [comment]
@peiying2 Oh, I thought they'd get into Pontoon automatically! (You're talking about the strings in this PR, right?) We'd like to put this in production September 13 I believe. Ideally we'd have most translations by the 6th, so we can have everything settle on the stage environment for a bit, but worst case we can do an l10n update just before going to production as well.

**Comment by peiying2** [comment]
> @peiying2 Oh, I thought they'd get into Pontoon automatically! (You're talking about the strings in this PR, right?) We'd like to put this in production September 13 I believe. Ideally we'd have most translations by the 6th, so we can have everything settle on the stage environment for a bit, but worst case we can do an l10n update just before going to production as well.

It lands in Pontoon automatically. If certain content must be ready by a deadline, we can add it to the file. It doesn't mean the deadline will be met, but it helps the communities prioritize the projects.

This is another benefit of separating these strings from the rest of the files, so high priority files can be worked on first. 
- https://pontoon.mozilla.org/it/ - deadline for each project due to release schedule - not a required field
- https://pontoon.mozilla.org/it/mozillaorg/ - file level priority and deadline (once the deadline has passed, it will be removed).

**Comment by Vinnl** [comment]
@peiying2 Hmm, I'm not sure I'm following... Are you referring to the strings in this PR, or are your comments in preparation of the work with limited availability outside the US and Canada?

For example, I'm not sure what you mean by "If certain content must be ready by a deadline, we can add it to the file." - is "the file" `en/app.ftl`? If so, didn't we just add the strings to that in this PR?

And thanks for the two links. So on that second link, are the new files automatically added to Pontoon? In other words, is it essentially free if we add more files than just `app.ftl`? Because if so, we might want to create separate files per feature always?

**Comment by peiying2** [comment]
@Vinnl we are good with this PR. 

My comment here is for the future. I want to make a case to have a separate file using the process we have for other projects. Mozilla.org has many files. When a page is tied to a release (like what's new page), we can add a deadline to that page only, not for the entire project. If a community has 200 strings to complete, they will work on these 10 strings first because it has a deadline. This would work for the promotion strings, not just prioritizing within this project, but against other projects that may have a due date. We can discuss more on Slack on how to approach this when the situation calls for it.

**Comment by Vinnl** [comment]
Gotcha! Yes, that makes sense, I didn't even know that was an option!

**Pull request merged**